sys/include/progress_bar.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_progress_bar
+ * @{
+ *
+ * @file
+ * @brief       A simple CLI progress bar
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef PROGRESS_BAR_H
+#define PROGRESS_BAR_H
+
+#include <stdlib.h>
+#include <inttypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Progress bar maximum characters length
+ */
+#ifndef PROGRESS_BAR_LENGTH
+#define PROGRESS_BAR_LENGTH                     (25U)
+#endif
+
+/**
+ * @brief   Progress bar character
+ */
+#ifndef PROGRESS_BAR_FULL_CHARACTER
+#define PROGRESS_BAR_FULL_CHARACTER              "█"
+#endif
+
+/**
+ * @brief   Progress bar empty character
+ */
+#ifndef PROGRESS_BAR_EMPTY_CHARACTER
+#define PROGRESS_BAR_EMPTY_CHARACTER            " "
+#endif
+
+/**
+ * @brief   Character displayed on the left of the progress bar
+ */
+#ifndef PROGRESS_BAR_PREFIX_CHARACTER
+#define PROGRESS_BAR_PREFIX_CHARACTER           "|"
+#endif
+
+/**
+ * @brief   Character displayed on the left of the progress bar
+ */
+#ifndef PROGRESS_BAR_SUFFIX_CHARACTER
+#define PROGRESS_BAR_SUFFIX_CHARACTER           "|"
+#endif
+
+/**
+ * @brief   Progress bar prefix max length
+ */
+#ifndef PROGRESS_BAR_PREFIX_MAX_LENGTH
+#define PROGRESS_BAR_PREFIX_MAX_LENGTH          (32U)
+#endif
+
+/**
+ * @brief   Progress bar suffix max length
+ */
+#ifndef PROGRESS_BAR_SUFFIX_MAX_LENGTH
+#define PROGRESS_BAR_SUFFIX_MAX_LENGTH          (32U)
+#endif
+
+/**
+ * @brief   Progress bar descriptor 
+ */
+typedef struct {
+    /**  Current value of the progress bar. Must be between 0 and 100 (included) */
+    uint8_t value;
+    /** Prefix displayed on the left of the progress bar */
+    char prefix[PROGRESS_BAR_PREFIX_MAX_LENGTH];
```

Comment:
Do you expect this to be dynamic? Using the array, this is forced to be in memory.
If you change this to be a plain `char*`, users can either allocate the memory themselves or pass a constant (nul-terminated) string.

Reply:
I tried this but it makes the usage less handy I think, and more error prone for the user.